### PR TITLE
Rbotha/feature 43 week 5 settings update

### PR DESCRIPTION
**Pull request by ReinhardtBotha**
**Details:**

- Message will be added to the HTML if geolocation and/or push notification permissions are not granted
- Function added to toggle push notification setting via API
- Code improvements added from Wonnyo's review
- Delete account added

**Tickets for this PR:**

- 42-week-5-user-profile-update
- 43-week-5-settings-update
- 76-week-5-user-profile-delete-account

### CODE REVIEW

**Review by whamester — CHANGES_REQUESTED** [review]
- We can remove the password form, I will create an endpoint to send a link to update the password, I triggered the reset password request for your account so you can check out how the flow works. This is the email that will receive the link: botha.wr@gmail.com (Check your spam folder)

- We have to validate which fields are being changed and only trigger the requests needed, so if the image has not changed, a request to the storage shouldn't be done.

- Add the alt attribute to the image
```
<img src="#" alt="#" id="profilePhoto" />
```
you can leave it generic or add a custom alt with javascript but it has to have content.

- You could change this function to receive the image URL, this is known as a pure function, so it doesn't depend on external variables but rather only on its parameters

```
function showProfilePic() {
	if (user && user.photo) {
		profilePhoto.setAttribute('src', user?.photo)
	} else {
		profilePhoto.setAttribute('src', '#')
	}
}

```

Suggestion
```
function showProfilePic(url = '#') {
	profilePhoto.setAttribute('src', url)
}
// Calling the function
showProfilePic(user?.photo)

```

Here we are using optional chaining, to safely access the user object properties, if the user doesn't exist or it doesn't have a photo, this will be undefined, and in the function definition, the param has a default value so in this case if undefined the fallback is the default value.

- If the image has been updated, first save it to the storage and then use the URL from the response to update the user object

saveUserInfo()
saveProfilePicture()

in other words, these two functions shouldn't be parallel, but saveUserInfo dependent on saveProfilePicture.

File: pages/settings/script.js [hunk 1]
```diff
@@ -1,84 +1,207 @@
...
+let userEmail = user?.email
+let userID = user?.id
+
+saveProfileInfoBtn.addEventListener('click', (e) => {
+	e.preventDefault()
+	userName = document.getElementById('name').value
+	userEmail = document.getElementById('email').value
+	user.name = userName
+	user.email = userEmail
+	setUserSession(user)
```
Comment (whamester) [inline pages/settings/script.js:30]: setUserSession(user) should be inside saveUserInfo(), because we only want to update the storage if the request to the endpoint is successful

File: pages/settings/index.html [hunk 2]
```diff
@@ -1,100 +1,109 @@
...
-      <h2>Are you sure you want to delete your account?</h2>
-      <button id="deleteAccountYesBtn">Yes</button>
-      <button id="deleteAccountNoBtn">No</button>
-    </div>
-  </section>
-</body>
-
-</html>
\ No newline at end of file
+			<div class="form-check form-switch">
```
Comment (whamester) [inline pages/settings/index.html:70]: Add some line breaks in this html document to separate the different sections.

File: pages/settings/script.js [hunk 3]
```diff
@@ -1,84 +1,207 @@
...
-dropArea.addEventListener("dragover", (e) => {
-  e.preventDefault();
-});
+showProfilePic()
 
-dropArea.addEventListener("drop", (e) => {
-  e.preventDefault();
+const dropArea = document.getElementById('dropArea')
+const inputFile = document.getElementById('inputImage')
+var picture
```
Comment (whamester) [inline pages/settings/script.js:80]: Is there a reason why have a `var` variable and not a `let`.
Also is a good idea to have your global variables at the top of the file

Comment (ReinhardtBotha) [reply pages/settings/script.js]: fixed in commit 3f39481

File: pages/settings/index.html (hunk 2 above)
Comment (ReinhardtBotha) [reply pages/settings/index.html:79]: addressed in commit [620c05e](https://github.com/whamester/WMDD-4885-Project/pull/15/commits/620c05eb17d52911dba2c460aef25ca60293acfe)

File: pages/settings/script.js (hunk 1 above)
Comment (ReinhardtBotha) [reply pages/settings/script.js]: fixed in commit [486adda](https://github.com/whamester/WMDD-4885-Project/pull/15/commits/486adda7af9ef34a8b6a28beef097f09df1c55e5)

File: pages/settings/index.html [hunk 4]
```diff
@@ -24,68 +24,72 @@
...
+		</div>
+
+		<h2>Your photo</h2>
+		<p>This will be displayed on your profile.</p>
+		<img src="#" alt="User profile picture" id="profilePhoto" />
+
+		<div id="fileDropWrapper">
+			<label for="inputImage" id="dropArea">
+				<input type="file" accept="image/*" id="inputImage" hidden />
+				<p>Drop your image here, or click to browse</p>
```
Comment (ALFAROSO) [inline pages/settings/index.html:57]: Maybe is better to use a <span> here instead of <p>

Comment (ReinhardtBotha) [reply pages/settings/index.html]: fixed in [649335e](https://github.com/whamester/WMDD-4885-Project/pull/15/commits/649335e40952afeb41a8595865ac3de6a6f70c71)

**Review by whamester — APPROVED** [review]
